weaviate/connection/rest.go

```diff
@@ -53,6 +54,38 @@ func NewConnection(scheme string, host string, httpClient *http.Client, headers
 	return connection
 }
 
+// WaitForWeaviate waits until weaviate is started up and ready
+// expects weaviat at localhost:8080
+func (con *Connection) WaitForWeaviate(startup_period int) error {
+	if startup_period < 0 {
+		return errors.New("'startup_period' needs to be an integer larger than zero")
+	}
+
+	for i := 0; i < startup_period; i++ {
```

Comment:
You could use tickers instead of the loop here. Otherwise we would wait for the request timeout and then sleep afterwards. SO we would wait longer than the startup period. (Might also be a good test?)

Examples https://gobyexample.com/tickers